### PR TITLE
Improve `escape_html_attr` performance

### LINKED ISSUE #4016: Utf16 handling in html attributes and text
### Describe the bug

#4014 follow up.

The `escape_html_attr()` helper currently escapes invalid utf16 code units (ie. unpaired surrogates). ~~Its sibling `escape_json_in_html()` doesn't.~~

Do we actually need to escape them? They are technically invalid according to the HTML spec, but treated as a soft-error, and apparently browsers will just treat them as unknown characters without any extra issues that I could find.

I guess this could also be affected by whatever platform we're running on, since the utf16 to utf8 conversion usually happens there (from what I could gather?). Do we know of any issues from passing invalid utf16 to any platform?

If there are no known issues, we could drop the escaping from attributes.
~~If there are, it's unlikely that they wouldn't also affect text nodes (ie. `<script>`'s contents), in that case we'd also need that on `escape_json_in_html`.~~ This is actually already the case.

If it turns out they're needed, I have a regex-based version ready that improves performance quite a bit over the character-by-character loop.

### Reproduction

n/a

### Logs

_No response_

### System Info

```shell
System:
    OS: Windows 10 10.0.19044
    CPU: (4) x64 Intel(R) Core(TM) i5-3570 CPU @ 3.40GHz
    Memory: 1.04 GB / 7.88 GB
  Binaries:
    Node: 17.4.0 - ~\scoop\apps\nodejs\current\node.EXE
    npm: 8.4.0 - ~\scoop\apps\nodejs\current\bin\npm.CMD
  Browsers:
    Chrome: 98.0.4758.102
    Edge: Spartan (44.19041.1266.0), Chromium (98.0.1108.50)
  npmPackages:
    @sveltejs/eslint-config: github:sveltejs/eslint-config#v5.8.0 => 5.8.0
```

### Severity

annoyance

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by mrkishi**
Closes #4016.
Depends on #4015.

Replaces the character-by-character loop with a regex-based `str.replace` call.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpx changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

File: packages/kit/src/utils/escape.js [hunk 1]
```diff
@@ -24,54 +24,43 @@ export function escape_json_in_html(val) {
...
-
 /** @type {Record<string, string>} */
 const escape_html_attr_dict = {
-	'<': '&lt;',
-	'>': '&gt;',
+	'&': '&amp;',
 	'"': '&quot;'
 };
 
+const escape_html_attr_regex = new RegExp(
```
Comment (benmccann) [inline packages/kit/src/utils/escape.js:33]: can we add a comment about why this is needed?

Comment (PH4NTOMiki) [reply packages/kit/src/utils/escape.js:33]: Why delete < and >
they break HTML if not escaped?

Comment (PH4NTOMiki) [reply packages/kit/src/utils/escape.js:33]: and we fixed it with #3798 and #3804

Comment (mrkishi) [reply packages/kit/src/utils/escape.js:33]: @PH4NTOMiki That was about `escape_json_in_html`, this is about `escape_html_attr`.

@benmccann Sorry about the comments, the previous code didn't have them and I thought they weren't necessary. I'll add some.

**Review by Rich-Harris — APPROVED** [review]

**Pull request merged**